src/main/java/org/terracotta/statistics/derived/latency/Jsr107LatencyMonitor.java

```diff
@@ -42,40 +44,36 @@ public void end(long time, long latency, T result) {
     sampling.end(time, latency, result);
   }
 
-  @Override
-  public Long minimum() {return statistic.minimum();}
-
-  @Override
-  public Long maximum() {return statistic.maximum();}
-
   /**
    * @return The average in microseconds or 0 if it does not exist yet
    */
   @Override
-  public Double average() {
-    return jsr107Average();
-  }
-
-  /**
-   * @return The average in microseconds or 0 if it does not exist yet
-   */
-  public double jsr107Average() {
-    Double value = statistic.average();
-    if (value == null) {
+  public double average() {
+    LatencyAccumulator accumulator = statistic.get();
+    long count = accumulator.count();
+    if (count == 0) {
       //Someone involved with 107 can't do math
-      return 0d;  /**
-       * @return The average in microseconds or 0 if it does not exist yet
-       */
+      return 0d;
     } else {
       //We use nanoseconds, 107 uses microseconds
-      return value / 1000f;
+      return accumulator.total() / 1_000.0 / count;
     }
   }
 
+  @Override
+  public Long minimum() {
```

Comment:
Why Long not long?

Reply:
because there is no minimum if no operation has been called. This was the behavior we had already. min and max null if no values and average is NaN if no value since 0/0 == NaN

Reply:
ok